utils/port_utils.py

```
@@ -0,0 +1,278 @@
+def get_hyperv_excluded_ranges() -> list[tuple[int, int]]:
+    """返回 Hyper-V / WSL 保留端口区间列表（start, end）。
+
+    在非 Windows 或查询失败时返回空列表。
+    """
+    if sys.platform != "win32":
+        return []
+    try:
+        import subprocess
+
+        result = subprocess.run(
+            ["netsh", "interface", "ipv4", "show", "excludedportrange", "protocol=tcp"],
```

Ruff (0.15.1)

[error] Starting a process with a partial executable path

(S607)